packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -338,12 +341,12 @@ export function usePress(props: PressHookProps): PressResult {
         }
 
         let target = e.target as Element;
-        let shouldStopPropagation = triggerPressEnd(createEvent(state.target, e), 'keyboard', state.target.contains(target));
-        removeAllGlobalListeners();
 
+        let shouldStopPropagation = triggerPressEnd(createEvent(state.target, e), 'keyboard', state.target.contains(target));
         if (shouldStopPropagation) {
-          e.stopPropagation();
+          state.target?.dispatchEvent(new KeyboardEvent('keyup', {...e, bubbles: false}));
```

Comment:
@LFDanLu when do you anticipate this being hit? I'm not seeing it fail any of our current tests if I revert this line

Reply:
hm, could've swore that without it I wasn't getting the onKeyDown attached to a button firing but it is working locally as well.

Reply:
ah I remember now, I'm not sure why I removed the `e.stopPropagation` actually but IMO we should be stopping propagation of the keyup since usePress has handled it. The dispatch was there in order to try and get the `keyUp` on the button to still fire but it doesn't seem to work...

Reply:
well, it should only stop it for the Enter and Space keys, everything else should go through
we don't allow through all the pointerDown/Up etc when we handle it, so I don't think there is a problem if we block the keyUp for those two keys?
The test I wrote shows that keyUp coming through, I actually thought that was a bit weird.

Reply:
At the moment the wrapping if statement that has `isValidKeyboardEvent` should already limit this block to only Enter/Space so readding the stopPropagation back above should be fine. However, adding it back in makes it so that `keyUp` doesn't fire on a Button when pressing Enter/Space even with the dispatchEvent, IMO if possible we should have `keyUp` on the same element handling the press still fire for Enter/Space but then stop at that level.

Reply:
I tried this, unfortunately we fail some others as a result.
I checked against pointerUp, and that does make it out, so I think your argument that keyUp should make it at least as far as pointerUp has merit.

Reply:
Just so we don't forget: after talking about this with the rest of the team, I think we can get rid of the dispatchEvent and stop propagation here. usePress has historically not stopped propagation of the keydown event technically since it happened at the window level on a non-capturing listener. useKeyboard should stop propagation where we need it to happen, all other cases should have onKeyUp continue bubbling